aiida/tools/graph/graph_traversers.py

```diff
@@ -204,31 +226,31 @@ def traverse_graph(starting_pks, max_iterations=None, get_links=False, links_for
         linktype_list.append(linktype.value)
     filters_backwards = {'type': {'in': linktype_list}}
 
-    if not isinstance(starting_pks, (list, set, tuple)):
-        raise TypeError(f'starting_pks must be of type list, set or tuple\ninstead, it is {type(starting_pks)}')
-
-    if not starting_pks:
-        if get_links:
-            output = {'nodes': set(), 'links': set()}
-        else:
-            output = {'nodes': set(), 'links': None}
-        return output
+    if not isinstance(starting_pks, Iterable):  # pylint: disable=isinstance-second-argument-not-valid-type
+        raise TypeError(f'starting_pks must be an iterable\ninstead, it is {type(starting_pks)}')
 
     if any([not isinstance(pk, int) for pk in starting_pks]):
         raise TypeError(f'one of the starting_pks is not of type int:\n {starting_pks}')
     operational_set = set(starting_pks)
 
+    if not operational_set:
+        if get_links:
+            return {'nodes': set(), 'links': set()}
+        return {'nodes': set(), 'links': None}
+
     query_nodes = orm.QueryBuilder()
     query_nodes.append(orm.Node, project=['id'], filters={'id': {'in': operational_set}})
     existing_pks = set(query_nodes.all(flat=True))
     missing_pks = operational_set.difference(existing_pks)
-    if missing_pks:
+    if missing_pks and missing_callback is None:
         raise exceptions.NotExistent(
-            f'The following pks are not in the database and must be pruned before this   call: {missing_pks}'
+            f'The following pks are not in the database and must be pruned before this call: {missing_pks}'
         )
+    elif missing_pks and missing_callback is not None:
```

Comment:
you really don't like nesting, do you ;-)

Reply:
nope, the flatter the better 😆